### DISCUSSION TITLE
Table not created in directus - [FORBIDDEN] You don't have permission to access this.

### DISCUSSION THREAD

**Discussion by adrian-stephens** [directus/directus#7053] [Archive]
MySQL mariadb,  latest image.  Directus,  latest image as of 2021-07-28.

Directus recognises tables (i.e. lists them in collections and schema) created outside directus, but cannot access them.

It shows "[FORBIDDEN] You don't have permission to access this. " in a popup on any attempt to list the table, or modify the schema.

Using Directus to create the identical table (except name) works as expected.   I cannot see any difference in the tables in MySQL - they have the same fields and primary key definitions.

I'm running as admin.  I've tried accessing the database as mysql root and the more restricted database user with no difference.

Can anybody suggest what I'm doing wrong?

**Comment by rijkvanzanten** [comment]
Do you have schema caching enabled? https://docs.directus.io/reference/environment-variables/#cache

**Reply by adrian-stephens** [reply]
I left those settings at their defaults.

so CACHE_ENABLED should be false, and CACHE_SCHEMA should be true.

I destroyed the container and re-created it.   This had an effect.

For an externally created table with no foreign keys,  I was now able to hit the warning triangle to the right of its one field,
and directus now recognises that as an OK table,  no error messages.

However,  I have an externally created table with a foreign key (one to many).  This was detected as a varchar (which is correct),
but when I tried to click on the warning triangle and save the changes,  I got the original error message back.

I then set CACHE_SCHEMA to false and recreated the container.
One immediate change is that the externally created table which it was happy with previously is now marked with a warning triangle,
so directus has a persistant knowledge that it did not create this table.  When I visit the data-model page for that table, under "Display Template"
it shows a warning triangle "select a collection", but doesn't allow entry of anything into the "Collection*" box.  So I have no way to dismiss the
warning triangle.  Notwithstanding the warning triangle,  the table browses and inserts OK,  but does not allow editing of an entry.  I suspect it is
not picking up the primary key (a VARCHAR) correctly.

I still have the original problem on another externally created table,  which includes a foreign key field (VARCHAR). Browsing fails with the
forbidden message.   In the data model, attempting to dismiss the warning triangle on the key field and click the OK gives the same message
as does the field that contains a foreign key.

I think my best bet is to recreate the tables in directus.

--------------------

Update.   I tried recreating the tables,  and had lots of problems.   I think the system doesn't like primary keys being strings and being referred to in 1:n relationships from other tables.

Anyhow,  I decided not to proceed further as directus doesn't support ui generation beyond browsing and editing tables.  I need to generate forms with subforms contains editable tables.  I realise I could do this as a vue extension,  but I'm not sure I want to learn how to do that now.

**Reply by rijkvanzanten** [reply]
> I think the system doesn't like primary keys being strings and being referred to in 1:n relationships from other tables.

Would you be able to share a sql dump with a minimal reproduction we can use to reproduce/debug with?
